compiler/src/dotty/tools/dotc/transform/SymUtils.scala

```diff
@@ -259,6 +259,16 @@ object SymUtils:
         && self.owner.linkedClass.is(Case)
         && self.owner.linkedClass.isDeclaredInfix
 
+    /** Is symbol declared or inherits @experimental? */
+    def isExperimental(using Context): Boolean =
+      // TODO should be add `@experimental` to `class experimental` in PostTyper?
+      self.eq(defn.ExperimentalAnnot)
+      || self.hasAnnotation(defn.ExperimentalAnnot)
+      || (self.maybeOwner.isClass && self.owner.hasAnnotation(defn.ExperimentalAnnot))
+
+    def isNestedInExperimental(using Context): Boolean =
+      self.ownersIterator.drop(1).exists(_.hasAnnotation(defn.ExperimentalAnnot))
```

Comment:
@smarter it seems I'm forcing to much by checking the annotation here during PostTyper. Does that make sense? How could I try to avoid this call to `hasAnnotation` on potentially problematic symbols?

Reply:
it seems weird to me that this could cause an error after Typer, is it related to macro usage?